catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/ListApplication.java

```diff
@@ -123,9 +124,16 @@ public void init() {
             IOUtils.copy(attachmentInfo.getStream(), temporaryFileBackedOutputStream);
 
             for (Splitter splitter : lookupSplitters(attachmentInfo.getContentType())) {
-              if (attemptToSplitAndStore(
-                  response, listType, attachmentInfo, temporaryFileBackedOutputStream, splitter))
-                break;
+              try {
+                if (attemptToSplitAndStore(
+                    response, listType, attachmentInfo, temporaryFileBackedOutputStream, splitter))
+                  break;
+              } catch (StopSplitterExecutionException e) {
+                String exceptionMessage = "Attached files do not contain the correct mimetypes";
```

Comment:
This message seems pretty specific for the exception.  Why not just print the stack trace and let the code that threw it convey the reason as to why it failed?

Reply:
I agree. Lets just log e